src/common/exponent/exponentHelper.ts

```diff
@@ -0,0 +1,384 @@
+            .then(usingExponent => {
+                if (!usingExponent) {
+                    return Q.resolve<void>(void 0);
+                }
+                return this.commandExecutor.execute("npm uninstall react-native", { silent: true })
+                    .then(() =>
+                        this.commandExecutor.execute(`npm install react-native`, { silent: true }));
+            })
+            .then(() => {
+                this.dependencyPackage = ReactNativePackageStatus.FACEBOOK_PACKAGE;
+            });
+    }
+
+    /**
+     * Reads VSCODE_EXPONENT Settings file. If it doesn't exists it creates one by
+     * guessing which entrypoint and filename to use.
+     */
+    private readVscodeExponentSettingFile(): Q.Promise<any> {
+        let defaultSettings = {
+            "entryPointFilename": "",
+            "entryPointComponent": "",
+            "createOrOverrideExpJson": true,
+        };
+        // Only create a new one if there is not one already
+        return this.fileSystem.exists(this.dotvscodePath(VSCODE_EXPONENT_JSON))
+            .then((vscodeExponentExists: boolean) => {
+                if (vscodeExponentExists) {
+                    return this.fileSystem.readFile(this.dotvscodePath(VSCODE_EXPONENT_JSON), "utf-8")
+                        .then(function (jsonContents: string): Q.Promise<any> {
+                            return JSON.parse(jsonContents);
+                        });
+                } else {
+                    return this.getPackageName()
+                        .then(packageName => {
+                            // By default react-native uses the package name for the entry component. This is our safest guess.
+                            defaultSettings.entryPointComponent = packageName;
+                            this.entrypointComponentName = defaultSettings.entryPointComponent;
+                            return this.fileSystem.exists(this.pathToFileInWorkspace(DEFAULT_IOS_INDEX));
+                        })
+                        .then((indexIosExists: boolean) => {
+                            // If there is an ios entrypoint we want to use that, if not let's go with android
+                            defaultSettings.entryPointFilename = indexIosExists ? DEFAULT_IOS_INDEX : DEFAULT_ANDROID_INDEX;
+                            this.entrypointFilename = defaultSettings.entryPointFilename;
+                            return this.fileSystem.writeFile(this.dotvscodePath(VSCODE_EXPONENT_JSON), JSON.stringify(defaultSettings, null, 4));
+                        })
+                        .then(() => {
+                            return defaultSettings;
+                        });
+                }
+            });
+    }
+
+    /**
+     * Exponent sdk version that maps to the current react-native version
+     * If react native version is not supported it returns null.
+     */
+    private get exponentSdk(): Q.Promise<string> {
+        if (this.expSdkVersion) {
```

Comment:
I think that's probably fine, since I also agree that updating react-native isn't something you do a lot. It is worth making sure that it is documented somewhere though I think.